src/utils/Obs_ArrayHelper.cpp

```diff
@@ -276,3 +276,43 @@ std::vector<json> Utils::Obs::ArrayHelper::GetSceneTransitionList()
 
 	return ret;
 }
+
+struct EnumSourceFilterInfo {
+    size_t index;
+    std::vector<json> filters;
+};
+
+std::vector<json> Utils::Obs::ArrayHelper::GetSourceFilterList(obs_source_t *source)
+{
+    EnumSourceFilterInfo filterInfo;
+
+    auto filterEnumProc = [](obs_source_t *source, obs_source_t *filter, void *param) {
+        auto filterInfo = reinterpret_cast<EnumSourceFilterInfo*>(param);
+
+        json filterJson;
+        filterJson["filterEnabled"] = obs_source_enabled(filter);
+        filterJson["filterIndex"] = filterInfo->index++;
+        filterJson["filterKind"] = obs_source_get_id(filter);
+        filterJson["filterName"] = obs_source_get_name(filter);
+
+        OBSDataAutoRelease filterSettings = obs_source_get_settings(filter);
+        filterJson["filterSettings"] = Utils::Json::ObsDataToJson(filterSettings);
+
+        filterInfo->filters.push_back(filterJson);
+    };
+    obs_source_enum_filters(source, filterEnumProc, &filterInfo);
+
+    return filterInfo.filters;
+}
+
+std::vector<std::string> Utils::Obs::ArrayHelper::GetFilterKindList()
```

Comment:
nitpick: the order of these two utils can be reversed to match the order of similar utils (get kind list first)